webui/react/src/components/FilterForm/components/FilterForm.tsx

```diff
@@ -18,16 +19,20 @@ interface Props {
 
 const FilterForm = ({ formStore, columns, onHidePopOver }: Props): JSX.Element => {
   const scrollBottomRef = useRef<HTMLDivElement>(null);
-  const data = useObservable(formStore.formset);
-  const isButtonDisabled = data.filterGroup.children.length > ITEM_LIMIT;
+  const loadableData = useObservable(formStore.formset);
+  const data = Loadable.getOrElse(null, loadableData);
+  const isButtonDisabled = (data?.filterGroup?.children?.length ?? -1) > ITEM_LIMIT;
 
   const onAddItem = (formKind: FormKind) => {
+    if (!data) return;
     formStore.addChild(data.filterGroup.id, formKind);
     setTimeout(() => {
       scrollBottomRef?.current?.scrollIntoView({ behavior: 'smooth', block: 'end' });
     }, 100);
   };
 
+  if (!data) return <div className={css.base} />;
```

Comment:
Shouldn't this be a loading state? Or could you put the button itself into a loading state?